src/main/java/Hari.java

```diff
@@ -0,0 +1,535 @@
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
+/**
+ * Class representing a task management chatbot named Hari.
+ */
+class HandlerBot {
+
+    private static final String FOLDER_PATH = "." + File.separator + "data";
+    private static final String FILE_PATH = FOLDER_PATH + File.separator + "hari.txt";
+
+    /**
+     * Main class for the Hari chatbot.
+     */
+    public Task[] arrTasks; // To store tasks created by the user for easy retrieval and listing
+    public int counterTasks; // Counter for assumption that there are no more than 100 tasks
+
+    // Class object
+    public HandlerBot() {
+        arrTasks = new Task[100]; // Assumption that there are no more than 100 tasks
+        counterTasks = 0; // Counter for the number of tasks
+    }
+
+    // Class for task (called Task)
+    public class Task {
+        private String summary; // Description of tasks
+        private boolean completion; // To check if a task is or is not completed
+        private String taskType; // To identify the type of task
+        private LocalDate deadline; // Add LocalDate for deadline
+        private String timerStart; // Start time for Event tasks
+        private String timerEnd; // End time for Event tasks
+        private String deadlineStat; // Deadline for Deadline tasks
+
+        // Constructor for To Do tasks
+        public Task(String summary) {
+            this.summary = summary;
+            this.completion = false;
+            this.taskType = "T";
+            this.deadline = null;
+            this.timerStart = null;
+            this.timerEnd = null;
+        }
+
+        // Constructor for Deadline tasks
+        public Task(String summary, String deadlineStat) {
+            this.summary = summary;
+            this.completion = false;
+            this.taskType = "D";
+            this.deadline = parseDeadline(deadlineStat);
+            this.timerStart = null;
+            this.timerEnd = null;
+        }
+
+        // Constructor for Event tasks
+        public Task(String summary, String startTime, String endTime) {
+            this.summary = summary;
+            this.completion = false;
+            this.taskType = "E";
+            this.deadline = null;
+            this.timerStart = startTime;
+            this.timerEnd = endTime;
+        }
+
+        // Additional method to parse deadline string into LocalDate
+        private LocalDate parseDeadline(String deadlineStat) {
+            try {
+                // Parse the date string into a LocalDate object
+                return LocalDate.parse(deadlineStat, DateTimeFormatter.ofPattern("yyyy-MM-dd"));
+            } catch (Exception e) {
+                // Handle parsing exception (invalid date format)
+                System.out.println("Error when parsing deadline. Please use the format yyyy-MM-dd.");
+                return null;
+            }
+        }
+
+        // To mark as completed
+        public void completionMark() {
```

Comment:
I think the naming of methods should be verbs instead of nouns. So maybe something like 'markAsComplete' would be better.

Reply:
Thanks! Have amended my method names (as best as possible) to reflect the purpose of the function itself.